src/System.ComponentModel.Annotations/tests/ValidationAttributeTestBase.cs

```diff
@@ -0,0 +1,135 @@
+        }
+
+        [Fact]
+        public void ErrorMessage_Valid()
+        {
+            if (!RespectsErrorMessage || InvalidValues().Count() == 0)
+            {
+                return;
+            }
+            ErrorMessageSet_ReturnsOverridenValue(InvalidValues().First());
+            ErrorMessageNotSet_ReturnsDefaultValue(InvalidValues().First());
+            ErrorMessageSetFromResource_ReturnsExpectedValue(InvalidValues().First());
+        }
+        
+        public void ErrorMessageNotSet(Test test)
+        {
+            test.Attribute.ErrorMessage = null;
+            Assert.Throws(InvalidErrorMessage_Type, () => test.Attribute.Validate(test.Value, test.ValidationContext));
+        }
+        
+        public void ErrorMessageSet_ErrorMessageResourceNameSet(Test test)
+        {
+            test.Attribute.ErrorMessage = "Some";
+            test.Attribute.ErrorMessageResourceName = "Some";
+            Assert.Throws(InvalidErrorMessage_Type, () => test.Attribute.Validate(test.Value, test.ValidationContext));
+        }
+        
+        public void ErrorMessageResourceNameSet_ErrorMessageResourceTypeNotSet(Test test)
+        {
+            test.Attribute.ErrorMessageResourceName = "Some";
+            test.Attribute.ErrorMessageResourceType = null;
+            Assert.Throws(InvalidErrorMessage_Type, () => test.Attribute.Validate(test.Value, test.ValidationContext));
+        }
+        
+        public void ErrorMessageResourceNameNotSet_ErrorMessageResourceTypeSet(Test test)
+        {
+            test.Attribute.ErrorMessageResourceName = null;
+            test.Attribute.ErrorMessageResourceType = typeof(ErrorMessageResources);
+            Assert.Throws(InvalidErrorMessage_Type, () => test.Attribute.Validate(test.Value, test.ValidationContext));
+        }
+        
+        public void ErrorMessageSet_ReturnsOverridenValue(Test test)
+        {
+            test.Attribute.ErrorMessage = "SomeErrorMessage";
+
+            var validationResult = test.Attribute.GetValidationResult(test.Value, test.ValidationContext);
+            Assert.Equal("SomeErrorMessage", validationResult.ErrorMessage);
+        }
+        
+        public void ErrorMessageNotSet_ReturnsDefaultValue(Test test)
+        {
+            test.Attribute.GetValidationResult(test.Value, test.ValidationContext);
+        }
+        
+        public void ErrorMessageSetFromResource_ReturnsExpectedValue(Test test)
+        {
+            test.Attribute.ErrorMessageResourceName = "InternalErrorMessageTestProperty";
+            test.Attribute.ErrorMessageResourceType = typeof(ErrorMessageResources);
+
+            var validationResult = test.Attribute.GetValidationResult(test.Value, test.ValidationContext);
+            Assert.Equal("Error Message from ErrorMessageResources.InternalErrorMessageTestProperty", validationResult.ErrorMessage);
+        }
+    }
+
+    public class Test
```

Comment:
Could you make this a protected nested class inside ValidationAttributeTestBase?  That way it won't pollute the namespace for other unrelated test classes.... "Test" is a pretty generic name, otherwise. (In addition, I'd suggest "TestCase" rather than just "Test".)